src/sync.h

```diff
@@ -232,12 +244,26 @@ class SCOPED_LOCKABLE UniqueLock : public Base
 template<typename MutexArg>
 using DebugLock = UniqueLock<typename std::remove_reference<typename std::remove_pointer<MutexArg>::type>::type>;
 
-#define LOCK(cs) DebugLock<decltype(cs)> UNIQUE_NAME(criticalblock)(cs, #cs, __FILE__, __LINE__)
+// When locking a Mutex, require negative capability to ensure the lock
+// is not already held
+inline Mutex& MaybeCheckNotHeld(Mutex& cs) EXCLUSIVE_LOCKS_REQUIRED(!cs) LOCK_RETURNED(cs) { return cs; }
+inline Mutex* MaybeCheckNotHeld(Mutex* cs) EXCLUSIVE_LOCKS_REQUIRED(!cs) LOCK_RETURNED(cs) { return cs; }
+
+// When locking a GlobalMutex, just check we don't know it's locked
```

Comment:
Sorry for bothering, but it's hard for me to parse this comment (sure I need to improve my English).

Maybe, re-word it a bit?

Reply:
```suggestion
// When locking a GlobalMutex, just check it is not locked in the surrounding scope.
```

Reply:
Done; thanks @MarcoFalke for the suggestion so I didn't have to ask for one :)